block/retriever.go

```diff
@@ -116,6 +117,21 @@ func (m *Manager) ProcessNextDABatch(daMetaData *da.DASubmitMetaData) error {
 		return batchResp.Error
 	}
 
+	if len(batchResp.Batches) == 0 {
+		m.logger.Info("no blocks in batch from DA", "daHeight", daMetaData.Height)
+		return fmt.Errorf("no blocks in batch from DA: %d", daMetaData.Height)
+	}
+
+	lastBatch := batchResp.Batches[len(batchResp.Batches)-1]
+
+	if len(lastBatch.Blocks) == 0 {
+		m.logger.Info("no blocks in last batch from DA", "daHeight", daMetaData.Height)
+		return fmt.Errorf("no blocks in last batch from DA: %d", daMetaData.Height)
+	}
+
+	lastReceivedHeight := lastBatch.Blocks[len(lastBatch.Blocks)-1].Header.Height
+	types.LastReceivedDAHeightGauge.Set(float64(lastReceivedHeight))
```

Comment:
don't we need to set it after a successful apply?

Reply:
nvm. saw @srene 's comment.
eventhough I still think metrics are not debug data, and from metric perspective, makes more sense to see the last block we **synced** from DA/SL